edx/frontend-app-learner-portal-enterprise
# ENT-3505: Learner expiration experience

### Description
* added SubscriptionSummaryCard
* added SubscriptionExpirationWarningModal
* refactored Dashboard, OfferSummaryCard, and SidebarCard to match new UX
* removed alert that was previously used to indicate expired subscription plan
* updated DashboardSidebar tests, added tests for SubscriptionSummaryCard and SubscriptionExpirationWarningModal

Note: post-expiration experience is not covered in this PR, it will come later.

### Testing Instructions
* Ensure test data exists for an Enterprise Subscriptions learner, having an offer assigned to the learner is nice to have but can be faked
* Temporarily edit **license_manager/apps/subscriptions/admin.py** to move `expiration_date` from `read_only_fields` to `writable_fields`
* Login to the LP as a learner with an active subscription. Observe the Subscription Status card on the right sidebar.
* From Django admin, update the SubscriptionPlan's `expiration_date` to be less than 60 days into the future. 
* Reload the LP dashboard and observe that the expiration warning modal is shown and the Subscription Status card has been updated with an "Expiring" badge and the new expiration date.
* After dismissing the modal, reload the page and observe the expiration warning modal is not shown. 
* From Django admin, update the SubscriptionPlan's `expiration_date` to be in the past (expired). 
* Reload the LP dashboard and observe that the expiration modal is shown and the Subscription Status card has been updated with an "Expired" badge and the new expiration date. 
* After dismissing the modal, reload the page and observe the expiration modal is still shown. 

### Screenshots
**Active Subscription:**
![image](https://user-images.githubusercontent.com/58956521/97480384-b373db00-1929-11eb-994a-0af56d5d637f.png)

**Expiring Soon:**
![image](https://user-images.githubusercontent.com/58956521/97480595-03eb3880-192a-11eb-90ff-9f08d1cad299.png)
![image](https://user-images.githubusercontent.com/58956521/97342088-3b8baf00-185c-11eb-8f4a-e3a9749be095.png)

**Expired:**
![image](https://user-images.githubusercontent.com/58956521/97480108-5bd56f80-1929-11eb-873a-13da531753a1.png)
![image](https://user-images.githubusercontent.com/58956521/97341836-eea7d880-185b-11eb-9e40-fa745dbbb12b.png)

**With OfferSidebarCard:**
![image](https://user-images.githubusercontent.com/58956521/97481133-bcb17780-192a-11eb-8292-23a7348488e6.png)
